### PR TITLE
Fixes https://github.com/novus/nvd3/issues/823

### PR DESCRIPTION
**Pull request by liquidpele**
by getting the max value in scale instead of assuming [0] was the max when calculating the x offset for the axis label.

I did a little code cleanup too, based on warnings in my IDE.  

Also set the example to use axis labels there.

### CODE REVIEW

File: src/models/axis.js [hunk 1]
```diff
@@ -84,7 +85,7 @@ nv.models.axis = function() {
                         .attr('y', 0)
                         .attr('x', w/2);
                     if (showMaxMin) {
-                        var axisMaxMin = wrap.selectAll('g.nv-axisMaxMin')
```
Comment (robinfhu) [inline src/models/axis.js]: Why did you remove 'var'?

File: src/models/axis.js [hunk 2]
```diff
@@ -217,12 +217,13 @@ nv.models.axis = function() {
                      if(labelPadding > width) width = labelPadding;
                      });
                      */
+                    console.log('debug: ', scale.range());
```
Comment (robinfhu) [inline src/models/axis.js]: remove console.log

**Comment by robinfhu** [comment]
I think you need to add back 'var axisMaxMin', otherwise you create a global variable.

File: src/models/axis.js (hunk 1 above)
Comment (liquidpele) [inline src/models/axis.js]: meant to move the declaration to the top of the switch statement.  It was doing duplicate declarations.  Fixed now.

**Comment by robinfhu** [comment]
Thanks!
